[ERROR] [e.automation.internal.RuleEngineImpl] - Failed to execute rule 'RULE-NAME': null

After the Upgrade to OH4.0, some of my rules won´t run for the first time they are triggerd.
It seems like it´s mostly the case if a trigger condition is reused for a few times for different rules.
In the openhab.log are the following errors:

example:
`[ERROR] [e.automation.internal.RuleEngineImpl] - Failed to execute rule 'usv-6': null` or
`[ERROR] [e.automation.internal.RuleEngineImpl] - Failed to execute rule 'rasenmaeher-2': null`

an with trace enabled:

```
│2023-08-12 17:25:00.206 [DEBUG] [e.automation.internal.RuleEngineImpl] -                                                                                          │
│java.util.ConcurrentModificationException: null                                                                                                                   │
│       at java.util.HashMap.computeIfAbsent(HashMap.java:1221) ~[?:?]                                                                                             │
│       at org.openhab.core.automation.internal.RuleEngineImpl.getContext(RuleEngineImpl.java:1103) ~[?:?]                                                         │
│       at org.openhab.core.automation.internal.RuleEngineImpl.updateContext(RuleEngineImpl.java:1090) ~[?:?]                                                      │
│       at org.openhab.core.automation.internal.RuleEngineImpl.setTriggerOutputs(RuleEngineImpl.java:1079) ~[?:?]                                                  │
│       at org.openhab.core.automation.internal.RuleEngineImpl.runRule(RuleEngineImpl.java:992) ~[?:?]                                                             │
│       at org.openhab.core.automation.internal.TriggerHandlerCallbackImpl$TriggerData.run(TriggerHandlerCallbackImpl.java:87) ~[?:?]                              │
│       at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]                                                                          │
│       at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]                                                                                         │
│       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]                                   │
│       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]                                                                  │
│       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]                                                                  │
│       at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

this rule is defines in rules/usv.rules:

```
rule "USV Batteriewarnung EG"
when
	Item EG_USV_Status changed
then
    if ((EG_USV_Status.state == "OB DISCHRG") || (EG_USV_Status.state == "OB") || (EG_USV_Status.state == "CB")) {
    	executeCommandLine("/home/openhab/signal-api.sh", "Achtung: USV-EG läuft auf Batterie", "+123")
    }
    else {
        //alles gut
    }
end
```


or an even simpler one:

```
// Triggers:
// - When GEM_Regensensor changed

// context: rasenmaeher-2
sendCommand(GEM_Rasenmaeher_Regentimer, ON)

```

If the rule is triggered again, everything just works until i restart openhab. This does not affect every rule, but some of them.

## Expected Behavior
the rules should be executed without any error as in previous versions like 3.5.

## Current Behavior
the following entry appears in openhab.log:
`[ERROR] [e.automation.internal.RuleEngineImpl] - Failed to execute rule 'usv-6': null`



## Your Environment
latest openHAB 4.0.1 on Debian AMD64 via apt with Zulu-17 installed.

--- a/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleEngineImpl.java
+++ b/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleEngineImpl.java
@@ -178,7 +178,7 @@ public class RuleEngineImpl implements RuleManager, RegistryChangeListener<Modul
      * The context map of a {@link Rule} is cleaned when the execution is completed. The relation is
      * {@link Rule}'s UID to Rule context map.
      */
-    private final Map<String, Map<String, Object>> contextMap = new HashMap<>();
+    private final Map<String, Map<String, Object>> contextMap = new ConcurrentHashMap<>();
 
     /**
      * This field holds reference to {@link ModuleTypeRegistry}. The {@link RuleEngineImpl} needs it to auto-map

